oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/tagtranslator/TagTranslator.java

```diff
@@ -66,12 +74,44 @@ public TagTranslator(Connection conn) throws OSHDBKeytablesNotFoundException {
     EnumSet<TableNames> keyTables =
         EnumSet.of(TableNames.E_KEY, TableNames.E_KEYVALUE, TableNames.E_ROLE);
     for (TableNames table : keyTables) {
-      try {
-        this.conn.prepareStatement("select 1 from " + table.toString() + " LIMIT 1").execute();
+      try (Statement testTablePresentQuery = this.conn.createStatement()) {
+        testTablePresentQuery.execute("select 1 from " + table.toString() + " LIMIT 1");
       } catch (SQLException e) {
         throw new OSHDBKeytablesNotFoundException();
       }
     }
+
+    // create prepared statements for querying tags from keytables
+    try {
+      keyIdQuery = conn.prepareStatement(
+          "select ID from " + TableNames.E_KEY.toString() + " where KEY.TXT = ?;");
+      keyTxtQuery = conn.prepareStatement(
+          "select TXT from " + TableNames.E_KEY.toString() + " where KEY.ID = ?;");
+      valueIdQuery = conn.prepareStatement("select k.ID as KEYID,kv.VALUEID as VALUEID"
+          + " from " + TableNames.E_KEYVALUE.toString() + " kv"
+          + " inner join " + TableNames.E_KEY.toString() + " k on k.ID = kv.KEYID"
+          + " where k.TXT = ? and kv.TXT = ?;");
+      valueTxtQuery = conn.prepareStatement("select k.TXT as KEYTXT,kv.TXT as VALUETXT"
+          + " from " + TableNames.E_KEYVALUE.toString() + " kv"
+          + " inner join " + TableNames.E_KEY.toString() + " k on k.ID = kv.KEYID"
+          + " where k.ID = ? and kv.VALUEID = ?;");
+      roleIdQuery = conn
+          .prepareStatement("select ID from " + TableNames.E_ROLE.toString() + " where TXT = ?;");
+      roleTxtQuery = conn
+          .prepareStatement("select TXT from " + TableNames.E_ROLE.toString() + " where ID = ?;");
+    } catch (SQLException e) {
+      throw new OSHDBKeytablesNotFoundException();
```

Comment:
How is the TagTranslator used with the ohsome-api, does it create a new Translator for every request or does it reuse an existing one?

Reply:
> We tested for the existence of the tables above, right? As a **nitpicker** I would prefer a different error or at least some explanatory string passed to the exception?

True, this would trigger most likely in the case where the keytables database and tables are present, but would not have the correct columns. In which they could be called "corrupted" maybe…. It's a very unlikely error case in normal use, though.

Anyway, it's not really related to the issue this PR is trying to resolve. Can you perhaps open a separate ticket?

Reply:
> How is the TagTranslator used with the ohsome-api, does it create a new Translator for every request or does it reuse an existing one?

I'm not sure how this relates to the comment by @SlowMo24… Did you mean to post it as a separate question, @rtroilo ?

In any case: If I'm not mistaken, the ohsome-api ends up having one TagTranslator for "normal" use (i.e. translating incoming tags from request parameters), and many TagTranslators for data extraction (translating tags for the GeoJSON output): one per thread per request. The "normal" tag translator is reused for all requests, if I remember correctly, while the ones used for data extraction requests are thrown away after the respective "parent" request is completed.

Does that matter for this PR?